src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClientProvider.cs

```diff
@@ -27,20 +27,30 @@ public Factory()
             {
             }
 
+            [Obsolete(MefConstruction.FactoryMethodMessage, error: true)]
             public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
-                => new ServiceHubRemoteHostClientProvider(workspaceServices);
+            {
+                if (!RemoteHostOptions.IsServiceHubProcess64Bit(workspaceServices)
+                    || workspaceServices.Workspace is not VisualStudioWorkspace)
```

Comment:
I thought we weren't supposed to use c#9 yet :-)

Reply:
It's enabled and can be used for new code. The only request was to not apply it in batch as a Fix All operation. 😄